### PR TITLE
merge pull request

### PR DESCRIPTION
**Pull request by benjamine**
adding `repo.mergePull`
docs: https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button

note the unit test isn't really testing a succesful merge (but a failture to merge), given the destructive nature of that action I think that would require mocking of the github API, but that might be a big effort.

ps: this is a pull request to merge pull requests :open_mouth:

### CODE REVIEW

**Comment by benjamine** [comment]
from #278

File: github.js [hunk 1]
```diff
@@ -458,6 +458,26 @@
             });
          };
 
+         // Merge a specific pull request
+         // -------
+
+         this.mergePull = function(pull, message, cb) {
```
Comment (AurelioDeRosa) [inline github.js]: Since we're using already an object as the first parameter, instead of using  overloading I think it's better to add the optional message as a property of the object. If it isn't defined we can use a default message (e.g. empty string).

File: github.js [hunk 2]
```diff
@@ -458,6 +458,26 @@
...
+               cb = message;
+               message = '';
+            }
+
+            var data = {
+               sha: pull.sha || pull.head.sha,
+               commit_message: message
+            };
+
+            _request('PUT', repoPath + '/pulls/' + pull.number + '/merge', data, function(err, mergeResult, xhr) {
```
Comment (AurelioDeRosa) [inline github.js]: The anonymous function can be replaced by `cb`.

File: test/test.repo.js [hunk 3]
```diff
@@ -272,14 +273,35 @@ describe('Creating new Github.Repository', function() {
    it('should get pull requests on repo', function(done) {
       var repo = github.getRepo('michael', 'github');
 
-      repo.getPull(153, function(err) {
+      repo.getPull(153, function(err, pull) {
```
Comment (AurelioDeRosa) [inline test/test.repo.js]: Can you remove this change and move it into another PR? This is not related to the merge feature.

**Comment by AurelioDeRosa** [comment]
Hi @benjamine. Are you still interested in this PR?

**Comment by benjamine** [comment]
@AurelioDeRosa yes, fixing

**Comment by benjamine** [comment]
@AurelioDeRosa all fixed except for this comment that got lost while rebasing:
> Since we're using already an object as the first parameter, instead of using overloading I think it's better to add the optional message as a property of the object. If it isn't defined we can use a default message (e.g. empty string).
the reason for I added `message` as a separate (optional) argument is that this allows you to send as first paramenter a `pull` object obtained from `getPull` (which will probably the most common case, as you can't create that yourself, github forces you to send the head sha to avoid mistakes)
This is shown in the 2nd example I added to README.
Otherwise you'd have to modify the object in the middle (with a parameter that isn't really part of a pull request info).

File: test/test.repo.js [hunk 4]
```diff
@@ -425,6 +426,25 @@ describe('Creating new Github.Repository', function() {
       });
    });
 
+   it('should merge pull requests on repo', function(done) {
+      var repo = github.getRepo('michael', 'github');
+      var expectedDocUrl = 'https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button';
+
+      repo.getPull(153, function(err, pull) {
+         repo.mergePull(pull, function(err) {
```
Comment (AurelioDeRosa) [inline test/test.repo.js]: I think we should have a test for the success of the merge of a pull request, not the failure. At the moment the test doesn't verify if the method is working correctly.

**Comment by AurelioDeRosa** [comment]
I see your point @benjamine and it makes perfect sense. However, I know that in that context a user will not care about the change of the original object with an optional `message` property. Also, this will help me in the big method refactor I'm planning.

File: test/test.repo.js (hunk 4 above)
Comment (benjamine) [inline test/test.repo.js]: I agree, but given there's no mocking of the github API, I don't see a way to do merges in a repeatable, parallel-safe, way.
But maybe the test title could be negative then.

**Comment by benjamine** [comment]
besides side effects, it seems confusing to me, as user, to add a `pull.message = ...` when it's not PR message you're setting (it's a commit message for the merge of such PR).
unless it's called `pull.mergeCommitMessage`?

File: test/test.repo.js (hunk 4 above)
Comment (AurelioDeRosa) [inline test/test.repo.js]: At the moment we're not doing parallel tests anyway, so the idea here is to create a new PR and then merge it.

Comment (benjamine) [inline test/test.repo.js]: I meant parallel in that any user could be running them, locally, or in travis from their forks.
should this create a random commit to a new branch and merge it (with the potential to get conflicts if other is merged in between)?

Comment (AurelioDeRosa) [inline test/test.repo.js]: No conflict can happen as the repository used is different at every execution of the tests.

Comment (benjamine) [inline test/test.repo.js]: oh ok, good to know, well I'll copy from the create pull request test
